lib/improver/utilities/filename.py

```diff
@@ -81,7 +89,48 @@ def generate_file_name(cube, parameter=None):
             parameter = parameter.replace(char, '')
         parameter = parameter.replace('__', '_')
 
-    filename = '{}-{}-{}.nc'.format(
-        validity_time_string, forecast_period_string, parameter)
+    period_string = None
+    if include_period:
+        coord_units = {"forecast_period": "seconds",
+                       "time": "seconds since 1970-01-01 00:00:00"}
+        for coord_name in ["forecast_period", "time"]:
```

Comment:
It is fairly clear what this is doing but it wouldn't hurt to put a comment above explaining it.

Reply:
Comment added.